src/Resources/Resources/ActiveDirectory/Cmdlets/NewAzureADServicePrincipalCommand.cs

```diff
@@ -383,7 +391,12 @@ private void CreateSimpleServicePrincipal()
 
         private bool SkipRoleAssignment()
         {
-            return this.IsParameterBound(c => c.SkipAssignment) || (!this.IsParameterBound(c => c.Role) && !this.IsParameterBound(c => c.Scope));
+            return this.IsParameterBound(c => c.SkipAssignment) || (!this.IsParameterBound(c => c.Role) && !this.IsParameterBound(c => c.Scope) && !HasSubscription());
+        }
+
+        private bool HasSubscription()
+        {
+            return DefaultContext.Subscription?.Id != null;
```

Comment:
You checked `Id` instead of `Subscription` itself. Does it mean it's possible to have a subscription with a null id 🤔

Reply:
this is equivalent to:
return Default.Subscription!=null && Default.Subscription.Id!=null;